symplyphysics/laws/relativistic/proper_time_for_timelike_intervals.py

```diff
@@ -0,0 +1,43 @@
+from sympy import Eq
+from sympy.physics.units import speed_of_light
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    validate_input,
+    validate_output,
+    assert_equal,
+)
+
+# Description
+## In relativity, proper time along a timelike world line is defined as the time as measured by a clock following
+## that line. The proper time interval between two events on a world line is the change in proper time, which is 
+## independent of coordinates, and is a Lorentz scalar.
+
+# Law: Δτ = Δs / c
+## Δτ - proper time
+## Δs - spacetime interval between two events
+## c - speed of light
+
+# Conditions
+## - The interval is timelike, i.e. `Δs` is real
+
+proper_time = Symbol("proper_time", units.time, real=True)
+spacetime_interval = Symbol("spacetime_interval", units.length)
+
+law = Eq(proper_time, spacetime_interval / speed_of_light)
+
+
+@validate_input(spacetime_interval_=spacetime_interval)
+@validate_output(proper_time)
+def calculate_proper_time(
+    spacetime_interval_: Quantity,
+) -> Quantity:
+    _, imag = spacetime_interval_.scale_factor.as_real_imag()
+    try:
+        assert_equal(imag, 0)
+    except AssertionError as e:
+        raise ValueError("Interval must be real") from e
```

Comment:
Don't try and catch. Simply throw when imaginary part is not zero.

Reply:
Is it because it's thought as a bad coding practice or it's just not necessary?

Reply:
Former. We catch errors when dealing with external libraries. When working with own code, organizing code flow with exceptions is bad practice. So you should simply check:
```
if Im(value) != 0:
    raise ValueError()
```

Reply:
Yeah, I see. But due to rounding errors the imaginary part might not be exactly zero, right? That's why I wanted to use `assert_equal`. Perhaps, if this function worked more like `expr_equals`, which returns a `bool` we then plug into `assert`, we could use it here.

Reply:
I think we'll usually work in real numbers space or complex numbers space. When dealing with real numbers, we are unlikely to get imaginary part due to rounding errors. That's why I suggest to use exact equality.